src/bin/compress.rs

```diff
@@ -102,6 +102,12 @@ fn main() {
         "invs": step_results.iter().map(|inv| inv.json()).collect::<Vec<serde_json::Value>>(),
     });
 
-    std::fs::write(&args.out, serde_json::to_string_pretty(&out).unwrap()).unwrap();
-    println!("Wrote to {:?}",args.out);
+    let out_path = &args.out;
+    if let Some(out_path_dir) = out_path.parent() {
+        if !out_path_dir.exists() {
```

Comment:
The double if is a bit ugly but it's necessary because "if"s and "if let"s are disjoint. Rust has implemented a fix to this (see [here](https://github.com/rust-lang/rfcs/pull/2497)), but it's not in the stable branch yet.